modules/core/02-client/legacy/v100/genesis_test.go

```diff
@@ -0,0 +1,272 @@
+	// This will cause the consensus states created before the first time increment
+	// to be expired
+	suite.coordinator.IncrementTimeBy(7 * 24 * time.Hour)
+
+	// migrate store get expected genesis
+	// store migration and genesis migration should produce identical results
+	err := v100.MigrateStore(path1.EndpointA.Chain.GetContext(), path1.EndpointA.Chain.GetSimApp().GetKey(host.StoreKey), path1.EndpointA.Chain.App.AppCodec())
+	suite.Require().NoError(err)
+	expectedClientGenState := ibcclient.ExportGenesis(path1.EndpointA.Chain.GetContext(), path1.EndpointA.Chain.App.GetIBCKeeper().ClientKeeper)
+
+	migrated, err := v100.MigrateGenesis(codec.NewProtoCodec(clientCtx.InterfaceRegistry), &clientGenState, suite.coordinator.CurrentTime)
+	suite.Require().NoError(err)
+
+	// check path 1 client pruning
+	//	fmt.Println(migrated.ClientsConsensus)
+	for _, height := range path1PruneHeights {
+		for _, client := range migrated.ClientsConsensus {
+			if client.ClientId == path1.EndpointA.ClientID {
+				for _, consensusState := range client.ConsensusStates {
+					suite.Require().NotEqual(height, consensusState.Height)
+				}
+			}
+
+		}
+		for _, client := range migrated.ClientsMetadata {
+			if client.ClientId == path1.EndpointA.ClientID {
+				for _, metadata := range client.ClientMetadata {
+					suite.Require().NotEqual(ibctmtypes.ProcessedTimeKey(height), metadata.Key)
+					suite.Require().NotEqual(ibctmtypes.ProcessedHeightKey(height), metadata.Key)
+					suite.Require().NotEqual(ibctmtypes.IterationKey(height), metadata.Key)
+				}
+			}
+		}
+	}
+
+	// check path 2 client pruning
+	for _, height := range path2PruneHeights {
+		for _, client := range migrated.ClientsConsensus {
+			if client.ClientId == path2.EndpointA.ClientID {
+				for _, consensusState := range client.ConsensusStates {
+					suite.Require().NotEqual(height, consensusState.Height)
+				}
+			}
+
+		}
+		for _, client := range migrated.ClientsMetadata {
+			if client.ClientId == path2.EndpointA.ClientID {
+				for _, metadata := range client.ClientMetadata {
+					suite.Require().NotEqual(ibctmtypes.ProcessedTimeKey(height), metadata.Key)
+					suite.Require().NotEqual(ibctmtypes.ProcessedHeightKey(height), metadata.Key)
+					suite.Require().NotEqual(ibctmtypes.IterationKey(height), metadata.Key)
+				}
+			}
+
+		}
+	}
+	bz, err := clientCtx.JSONCodec.MarshalJSON(&expectedClientGenState)
+	suite.Require().NoError(err)
+
+	// Indent the JSON bz correctly.
+	var jsonObj map[string]interface{}
+	err = json.Unmarshal(bz, &jsonObj)
+	suite.Require().NoError(err)
+	expectedIndentedBz, err := json.MarshalIndent(jsonObj, "", "\t")
+	suite.Require().NoError(err)
+
+	bz, err = clientCtx.JSONCodec.MarshalJSON(migrated)
+	suite.Require().NoError(err)
+
+	// Indent the JSON bz correctly.
+	err = json.Unmarshal(bz, &jsonObj)
+	suite.Require().NoError(err)
+	indentedBz, err := json.MarshalIndent(jsonObj, "", "\t")
+	suite.Require().NoError(err)
+
+	fmt.Println(string(indentedBz))
```

Comment:
the tests I followed printed the actual. I can remove, but it is useful in quickly verifying things